src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/MFDManipulation.java

```diff
@@ -79,7 +79,7 @@ public static UncertainIncrMagFreqDist addMfdUncertainty(IncrementalMagFreqDist
         EvenlyDiscretizedFunc stdDevs = new EvenlyDiscretizedFunc(mfd.getMinX(), mfd.getMaxX(), mfd.size());
         for (int i = 0; i < stdDevs.size(); i++) {
             double rate = mfd.getY(i);
-            double stdDev = ((i < minMagBin) || (maxMagBin < i))? 1e-20 : firstWeightPower / Math.pow(rate, power - 1);
+            double stdDev = ((i < minMagBin) || (maxMagBin < i) || rate == 1e-20 )? 1e-20 : firstWeightPower / Math.pow(rate, power - 1);
```

Comment:
Add TODO to remove 1e-20 when we're happy

Reply:
done